specs/agents/universal-profiling-integration.md

```diff
@@ -212,3 +212,54 @@ count                 | uint16
 * *transaction-id*: The APM W3C span id of the transaction which was active for the given profiling samples
 * *stack-trace-id*: The unique ID for the stacktrace captured assigned by the profiler. This ID is stored in elasticsearch in base64 URL safe encoding by the universal profiling solution.
 * The number of samples observed since the last report for the (*trace-id*, *transaction-id*, *stack-trace-id*) combination.
+
+
+# APM Agent output of correlation data
+
+## Correlation Attribute
+
+APM Agents will receive the IDs of stacktraces which occurred during transactions via [correlation messages](#cpu-profiler-trace-correlation-message).
+If the correlation feature is enabled, agents SHOULD store the received IDs of stacktraces as `elastic.profiler_stack_trace_ids` OpenTelemetry span attribute on the transaction:
+
+ * The type of the `elastic.profiler_stack_trace_ids` MUST be string-array
+ * The stacktrace-IDs MUST be provided as base64 URL-safe encoded strings without padding
+ * The order of elements in the array is not relevant
+ * The counts of stacktrace-IDs must be preserved: If a stacktrace occurred *n* times for a given transaction, its ID must appear exactly *n* times in `elastic.profiler_stack_trace_ids` of that transaction
+
+The APM intake will store `elastic.profiler_stack_trace_ids` as `transaction.profiler_stack_trace_ids` on transaction documents with the special `counted_keyword` mapping type, ensuring duplicates are preserved.
+
+For example, if for a single transaction the following correlation messages are received
+
+* (stack-trace-ID: 0x60b420bb3851d9d47acb933dbe70399b, count: 2)
+* (stack-trace-ID: 0x4c9326bb9805fa8f85882c12eae724ce, count: 1)
+* (stack-trace-ID: 0x60b420bb3851d9d47acb933dbe70399b, count: 1)
+
+the resulting transaction MUST have the OpenTelemetry attribute `elastic.profiler_stack_trace_ids` with a value of `[YLQguzhR2dR6y5M9vnA5mw, YLQguzhR2dR6y5M9vnA5mw, TJMmu5gF-o-FiCwS6uckzg, YLQguzhR2dR6y5M9vnA5mw]`.
```

Comment:
@JonasKunz This sounds as if the order of the stacktrace ids is relevant. Is this really the case? And if so, why?

Reply:
A few lines above I've listed `The order of elements in the array is not relevant` to avoid this confusion. Did you overlook this statement or is it not clear enough here in context of the example?

I just kept the array in the example in the same order as the messages for simplicity.

Reply:
Indeed, I overlooked it (was just skimming the text).
What triggered me was the `MUST` in `the resulting transaction MUST have the OpenTelemetry attribute` that my brain extended to `with a value of`.

My suggestion would be something like `with a value of (elements in any order)`. It may be duplicated information to you, but makes it very clear to the random reader (and discharges the `MUST`).

Reply:
Agree with you, let's make this more obvious with your suggested change. I've opened #864 to fix it.